src/Microsoft.IdentityModel.Tokens/SecurityTokenDescriptor.cs

```
@@ -116,5 +117,58 @@ public class SecurityTokenDescriptor
         /// </summary>
         [DefaultValue(true)]
         public bool IncludeKeyIdInHeader { get; set; } = true;
+
+        private static int s_maxActorChainLength = 5;
+        /// <summary>
+        /// Gets or sets the maximum depth allowed when processing nested actor tokens.
+        /// This prevents excessive recursion when handling deeply nested actor tokens.
+        /// The value must be at least 0. Value 0 would mean that the actor token is not allowed to be nested.
+        /// Default value is 5. Max value is also 5
+        /// </summary>
+        /// <exception cref="ArgumentOutOfRangeException">Thrown if the value is less than 0.</exception>
+        public static int MaxActorChainLength
+        {
+            get => s_maxActorChainLength;
+            set
+            {
+                if (value < 0)
```

Consider failing early by also checking if value is <= 5.